aiohttp/web_request.py

```diff
@@ -21,6 +22,35 @@
 FileField = collections.namedtuple(
     'Field', 'name filename file content_type headers')
 
+_TCHAR = string.digits + string.ascii_letters + r"!#$%&'*+\-.^_`|~"
+# notice the escape of '-' to prevent interpretation as range
+
+_TOKEN = r'[{tchar}]*'.format(tchar=_TCHAR)
+
+_QDTEXT = r'[{}]'.format(
+    r''.join(chr(c) for c in (0x09, 0x20, 0x21, *range(0x23, 0x7F))))
+# qdtext includes 0x5C to escape 0x5D ('\]')
+# qdtext excludes obs-text (because obsoleted, and encoding not specified)
+
+_QUOTED_PAIR = r'\\[\t {tchar}]'.format(tchar=_TCHAR)
+
+_QUOTED_STRING = r'"(?:{quoted_pair}|{qdtext})*"'.format(
+    qdtext=_QDTEXT, quoted_pair=_QUOTED_PAIR)
+
+_FORWARDED_PARAMS = (
+    r'[bB][yY]|[fF][oO][rR]|[hH][oO][sS][tT]|[pP][rR][oO][tT][oO]')
+
+_FORWARDED_PAIR = (
+    r'^ *({forwarded_params})=({token}|{quoted_string}) *$'.format(
```

Comment:
I'm not sure but as far as I can see that's not the case. Section 4 only says `forwarded-pair = token "=" value` and section 7 recalls `Note that an HTTP list allows white spaces to occur between the identifiers, and the list may be split over multiple header fields.`